hayeon/programmers-1/신고-결과-받기.js

```diff
@@ -0,0 +1,36 @@
+function solution(id_list, report, k) {
+  const answer = new Array(id_list.length);
+  answer.fill(0);
+
+  const report_list = {};
+
+  id_list.map((user) => {
+    report_list[user] = [];
+  });
+
+  report.map((user) => {
+    const [user_id, report_id] = user.split(" ");
+
+    if (!report_list[report_id].includes(user_id)) {
+      report_list[report_id].push(user_id);
+    }
+  });
+
+  for (const key in report_list) {
+    if (report_list[key].length >= k) {
+      report_list[key].map((user) => {
+        answer[id_list.indexOf(user)] += 1;
+      });
+    }
+  }
+  return answer;
+}
```

Comment:
- 풀이하느라 고생하셨씁니다 ~ 🙇🏻‍♂️
- 로직은 잘 짜셨는데, 기본적인 자바스크립트 네이밍 규칙과 forEach, map메서드의 용도에 맞는 사용 등 개선할게 많습니다..🥲
- 해당 문제에서는 입력값으로 id_list라고 넘어오긴하지만 저희가 새로운 변수를 만들고 활용할 때는 기본적인 카멜 케이스 네이밍으로 사용해주세요!!
```suggestion
function solution(id_list, report, k) {
  // 밑에다 쓸 필요없이 바로 메서드 체이닝 형태로 작성해도 됌
  const answer = new Array(id_list.length).fill(0);
  // 배열이 아닌데 list라는 네이밍과 네이밍 스네이크 -> 카멜 케이스 개선
  const reportObj = {}; 
    
  // 새로운 배열을 반환하는게 아니라 map이 아닌 forEach (1)
  id_list.forEach((user) => {
    reportObj[user] = [];
  });

  // 새로운 배열을 반환하는게 아니라 map이 아닌 forEach (2)
  report.forEach((user) => {
    // 스네이크 케이스 -> 카멜 케이스로 변환
    // 신고 당한 사람의 ID를 의미하는 reportId -> reportedUserId로 변환
    const [userId, reportedUserId] = user.split(" ");

    if (!reportObj[reportedUserId].includes(userId)) {
      reportObj[reportedUserId].push(userId);
    }
  });
    
  for (const userId in reportObj) {      
    if (reportObj[userId].length >= k) {
      // 새로운 배열을 반환하는게 아니라 map이 아닌 forEach (3)
      reportObj[userId].forEach((user) => {
        answer[id_list.indexOf(user)] += 1;
      });
    }
  }
    
  return answer;
}
```

Reply:
넵넵 감사합니다!!!! 주석으로 풀이까지 남겨주셔서 감사해요!! 아직도 객체로 푸는게 좀 어려운거 같습니다.. 좀더 노력해보겠습니다!!